### PR TITLE
Write demo - API and Morello

### PR DESCRIPTION
**Pull request by jonmattgray**
- Write Demo sends the full `[usernamePasswordPairs]` to API access-demo on every login attempt. **Login attempts are saved and run in order with each new login appended so users can exploit the out of bounds write bug.**
- Displays spinner whilst waiting for request.
- Shows text on login fail. Shows 'secret desktop' on login successful (same on both 'old' and 'new' style)
- After successfully login displays button to switch to Morello.
- If hack attempted on Morello, by logging in with an overflowing `username`, shows text to say user is 'locked out' as Morello throws an error. Any further login attempts will always show they are locked out because the previous hack attempt will be run first.
- After any login attempt on Morello, displays learn more button.

Spinner
image

Unsuccessful login
image

Secret desktop
image

**MORELLO**
Unsuccessful
image

Secret desktop
image

Hack!
image

### CODE REVIEW

File: cypress/integration/write-demo.spec.js [hunk 1]
```diff
@@ -35,19 +35,41 @@ describe('Write Demo', () => {
     })
   })
 
-  describe('Happy path', () => {
-    beforeEach(() => {
-      cy.get('[data-cy=username]').type('root')
-      cy.get('[data-cy=password]').type('password')
-      cy.get('[data-cy=login]').click()
-    })
+  it.skip('Shows login failed message after failed login', () => {
```
Comment (mattdean-digicatapult) [inline cypress/integration/write-demo.spec.js:38]: skipped?

File: src/components/demos/write/Demo.js [hunk 2]
```diff
@@ -15,129 +20,94 @@ const Wrapper = styled.div`
...
+                  showSpinner={awaitingApi}
+                  setUsernamePasswordPairs={setUsernamePasswordPairs}
+                  apiOutput={apiOutput}
+                />
               </Container>
-            </form>
-          </Container>
-        </Box>
+            </Box>
+            {isMorello && apiOutput && (
```
Comment (mattdean-digicatapult) [inline src/components/demos/write/Demo.js:101]: I think this should only be triggered once the hack has either been successful or when the hack has been prevented by cheri

File: src/components/demos/write/LoginForm.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+          {...demoState.theme.form.loginAttempt}
+          visibility={
+            failedLogin(apiOutput) || loginError(apiOutput)
+              ? 'visible'
+              : 'hidden'
+          }
+          data-cy={'login-attempt'}
+        >
+          {failedLogin(apiOutput) && `Incorrect username or password`}
+          {loginError(apiOutput) && `Hack attempt detected - locked out!`}
```
Comment (mattdean-digicatapult) [inline src/components/demos/write/LoginForm.js:118]: I think the copy and styling could be better on this but it's the right approach

File: cypress/integration/write-demo.spec.js (hunk 1 above)
Comment (jonmattgray) [reply cypress/integration/write-demo.spec.js:38]: was putting them as placeholders in anticipation of API integration testing being completed https://digicatapult.atlassian.net/browse/DSBD-34 

but I can remove, or change them to not be skips (or actually write get the API calls working in the tests!)

File: src/components/demos/write/Demo.js (hunk 2 above)
Comment (jonmattgray) [reply src/components/demos/write/Demo.js]: added https://github.com/digicatapult/morello-ui/pull/30/commits/b0724eb19a28836599799fb48f0279eaca01548a

File: src/components/demos/write/LoginForm.js (hunk 3 above)
Comment (jonmattgray) [reply src/components/demos/write/LoginForm.js]: changed the hack attempt one to `Suspicious activity detected - account locked`

Comment (jonmattgray) [reply src/components/demos/write/LoginForm.js]: and fixed some broken font css so fonts display correctly for Morello - see updated screenshots in PR description

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**